apex/contrib/torchsched/inductor/event.py

```diff
@@ -0,0 +1,182 @@
+
+    Args:
+        factory: The CUDAEventFactory that generate this event.
+        idx: Indexing number assigned in chronological order during scheduling.
+        ref_count: Reference count of this event instance.
+        materialized_event: The actual CUDA Event name that will be used in the final PyTorch
+            program. Only symbolic event with reference count larger than one will be materialized.
+
+    Note:
+        In most cases this class should not be used standalone. Use
+        `CUDAEventFactory.get_sym_event()` to instantiate one.
+    """
+
+    factory: CudaEventFactory
+    idx: int
+    ref_count: int = 0
+    materialized_event: str | None = None
+
+    def __lt__(self, rhs: CudaEventSym) -> bool:
+        """Whether the current event is generated before the rhs event."""
+        return self.idx < rhs.idx and self.factory is rhs.factory
+
+    def __eq__(self, rhs: object) -> bool:
+        """Whether the current event is identical to the rhs event."""
+        if not isinstance(rhs, CudaEventSym):
+            return NotImplemented
+        return self.idx == rhs.idx and self.factory is rhs.factory
+
+    def __str__(self) -> str:
+        """Represent this symbolic event in string."""
+        ret = f"{self.__class__.__name__} (idx={self.idx}"
+        if self.ref_count:
+            ret += f", ref_count={self.ref_count}"
+        if self.materialized_event:
+            ret += f", materialized to `{self.materialized_event}`"
+        ret += ")"
+        return ret
+
+    def __hash__(self) -> int:
+        """Hash this symbolic event."""
+        return hash(f"{id(self.factory)=},{self.idx=}")
+
+    def record(self, stream_idx: int) -> _CudaEventRecordLine:
+        """Record this event on a given stream.
+
+        Args:
+            stream_idx: The index of the stream that this event will record on.
+
+        Returns:
+            An internal data structure that depicts stream <-> event dependency.
+
+        Note:
+            This method doesn't directly generate a event recording in the final program. Instead
+            it records the dependence between the stream and the current event. Whether or not this
+            event recording show up in the final program depends on the reference count of the
+            current event. I.e., if this event is never waited for by the later code, this event
+            recording will not be code-generated.
+        """
+        stream = get_stream_name(stream_idx)
+        return _CudaEventRecordLine(self, stream)
+
+    def wait(self, stream_idx: int) -> _CudaEventWaitLine:
+        """Wait for this event to complete by a given stream.
+
+        Args:
+            stream_idx: The index of the stream that will be waiting for this event to complete.
+
+        Returns:
+            An internal data structure that depicts stream <-> event dependency.
+
+        Note:
+            This method doesn't directly generate a event waiting in the final program. Instead
```

Comment:
The wording here is confusing, is "necessarily" a more accurate description rather than "directly?"